docs/proposals/oras-attach-multi-platform.md

```diff
@@ -0,0 +1,91 @@
+
+## Scenarios
+
+### Attach an End-of-Life (EoL) Annotation as a Referrer to a Multi-Platform Image
+
+A security engineer, Alice, wants to use annotations to store EoL metadata to indicate that an image is no longer valid. Consider a multi-platform image `demo/alpine:a1a1` with multiple platform-specific images. When a vulnerability is detected in a platform-specific image `demo/alpine:b1b1`, it is patched, generating a new digest. The outdated image is marked as invalid using an EoL annotation:
+
+```json
+"vnd.demo.artifact.lifecycle.end-of-life.date": "2025-03-20T01:20:30Z"
+```
+
+Since the patched image has a new digest, the parent image index also receives an updated digest. When a platform-specific image is marked as EoL, dependent services stop using it, and vulnerability scanning tools can recognize it as deprecated through the lifecycle metadata.
+
+After patching `demo/alpine:b1b1`, Alice has to manually retrieve the new digests and run `oras attach` twice: once for the old digest of the parent image index and once for the platform-specific image manifest. If multiple platform-specific images require updates, she has execute multiple commands.
+
+```console
+demo/alpine:a1a1 (image index)
+-> demo/alpine:b1b1  <-- VULNERABLE
+-> demo/alpine:c1c1
+```
+
+After patching, Alice attaches the EoL annotation to the outdated image index and platform-specific image:
+
+```sh
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z"
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z" --platform linux/amd64
+```
+
+Resulting image structure:
+
+```console
+demo/alpine:a1a1 (image index) <-- VULNERABLE, EoL attached
+-> demo/alpine:b1b1 (linux/amd64)  <-- VULNERABLE, EoL attached
+-> demo/alpine:c1c1 (linux/arm64)
+```
+
+A new image index is created:
+
+```console
+demo/alpine:z1z1 (image index)
+-> demo/alpine:r1r1 (linux/amd64)
+-> demo/alpine:c1c1 (linux/arm64)
+```
+
+In addition, if a vulnerability is detected and affects all platform-specific images, the parent index and each platform-specific image are patched, generating new digest of each. The outdated multi-platform image and each platform-specific image are marked as invalid using an EoL annotation similar as above.
+
+```console
+demo/alpine:a1a1 (image index)
+-> demo/alpine:b1b1  <-- VULNERABLE
+-> demo/alpine:c1c1  <-- VULNERABLE
+```
```

Comment:
This scenario can be summarized as "downward propagation". That is, updating one node (e.g. the index) will cascading updates all child nodes.

Reply:
Here we focus on the image instead of the platform since the vulnerability is not platform specific.

Reply:
Updated based on your suggestion